File: dcp-solutions/2024-11-17/solution.py
```python
def count_bishop_attacks(n, bishops):
    attacks = set()
    bishop_set = set(bishops)
    
    def get attacking_diagonal(row, col):
        direction = 1 if row + col < n else -1
        diagonal = (row + direction) * direction
        return diagonal

    for i, bishop in enumerate(bishops):
        for j in range(i+1, len(bishops)):
            other_bishop = bishops[j]
            if abs(get_attacking_diagonal(bishop[0], bishop[1]) - get_attacking_diagonal(other_bishop[0], other_bishop[1])) == abs(bishop[0] - other_bishop[0]):
                attacks.add(tuple(sorted((i, j))))
                
    return len(attacks)
```

File: dcp-solutions/2024-11-17/solution.ex
```elixir
def count_attacking_bishops(m, bishops) do
  bishops_map = Map.new()
  for {row, col} <- Enum.map(bishops, fn {x, y} -> {rem(x, m), rem(y, m)} end),
      do: bishops_map[Map.put(bishops_map, {rem(x, m), rem(y, m)}, nil), row] = [row]

  attacking_pairs = Enum.flat_map(bishops_map,
    fn {diagonal, rows} ->
      Enum.map(rows, fn row -> Map.get(bishops_map, diagonal, []) ++ [{row, diagonal}] end)
    end)

  Enum.reduce(attacking_pairs, 0, &(&1 + &2 - 1))
end
```